### DISCUSSION TITLE
Is a Data API coming?

### DISCUSSION THREAD

**Discussion by tlaverdure** [cockroachdb/cockroach#101009] [Q&A]
Does Cockroach plan to add a data api that allows SQL queries over HTTP?

**Comment by Yash-xoxo** [comment]
CockroachDB doesn't have a built-in HTTP data API currently, but you have options:

**What exists:**
- **HTTP Admin API** - cluster management, not SQL queries
- **SQL over pgwire** - standard Postgres protocol (TCP)

**Workarounds:**

1. **PostgREST** - Auto-generates REST API from your schema
```bash
   docker run postgrest/postgrest
   # Point at CockroachDB as Postgres
```

2. **Hasura** - GraphQL API over Postgres-compatible DBs
   
3. **Roll your own** - Simple HTTP wrapper:
```go
   http.HandleFunc("/query", func(w http.ResponseWriter, r *http.Request) {
       db.Query(r.FormValue("sql"))
   })
```

**Why no native HTTP API?**
- CockroachDB focuses on wire protocol compatibility
- HTTP adds overhead vs pgwire
- Ecosystem tools like PostgREST handle this well

**Feature request:** If you want native support, open an issue - the team tracks demand for features like this.

For serverless-style HTTP SQL access, consider **CockroachDB Serverless** which has better connection pooling for HTTP-like workloads.
